src/Esprima/Loc.cs

```diff
@@ -74,16 +107,125 @@ public override string ToString()
         return Source is not null ? interval + ": " + Source : interval;
     }
 
-    public static bool operator ==(in Location left, in Location right)
+    private static bool TryParse(ReadOnlySpan<char> s, bool throwIfInvalid, out Location result)
     {
-        return left.Equals(right);
+        int i;
+        if (s[0] != '[' || (i = s.IndexOf(')')) < 0 || ++i < 5)
+        {
+            goto InvalidFormat;
+        }
+
+        var source = s.Slice(i);
```

Comment:
no sure if slicer is best because it's optimized for character strings, this is numeric, but probably not in hot path

Reply:
Hmm, I'm not sure I get this. This `Slice` is not our pooling slice but a plain old ROS `Slice`...

Reply:
Ah sorry, my bad. Span it is.